astropy/config/tests/test_configs.py

```diff
@@ -118,6 +118,56 @@ def test_set_temp_cache_resets_on_exception(tmp_path):
     assert t == paths.get_cache_dir()
 
 
+def _cleanup_environment(env_var_name: str, old_value: str | None) -> None:
+    if old_value is None:
+        os.environ.pop(env_var_name)
+    else:
+        os.environ[env_var_name] = old_value
```

Comment:
Why are you implementing this functionality yourself instead of making use of [`pytest` monkeypatching](https://docs.pytest.org/en/8.0.x/how-to/monkeypatch.html#monkeypatching-environment-variables)?

Reply:
I try to reserve `mock`/`monkeypatching` as last resorts because I think they are not well understood and tend to effectively make tests harder to read or refactor (see for instance https://github.com/astropy/astropy/pull/16659).